chain/chain/src/chain.rs

```diff
@@ -5512,6 +5515,10 @@ impl<'a> ChainUpdate<'a> {
                         .save_epoch_light_client_block(&prev_epoch_id.0, light_client_block);
                 }
             }
+
+            let shard_layout = self.epoch_manager.get_shard_layout_from_prev_block(prev.hash())?;
+            SHARD_LAYOUT_VERSION.set(shard_layout.version() as i64);
+            SHARD_LAYOUT_NUM_SHARDS.set(shard_layout.num_shards() as i64);
```

Comment:
When reviewing it's good to think about the placement of those lines. I placed it in postprocess_block, after the head is updated and in the metrics I report the values of the new head.

Reply:
Should we have this emit only when `self.epoch_manager.will_shard_layout_change(parent_hash)` is true?

Reply:
Why are we emitting this for every block processed?

Reply:
Yeah, I think so, it's only two numbers per block.